packages/mds-geography-author/api.ts

```diff
@@ -0,0 +1,292 @@
+
+function api(app: express.Express): express.Express {
+  app.get(
+    pathsFor('/geographies/meta/'),
+    checkAccess(scopes => {
+      return scopes.includes('geographies:read:published') || scopes.includes('geographies:read:unpublished')
+    }),
+    async (req, res) => {
+      // TODO
+      // if get_unpublished is true, and client lacks geographies:read:published, then
+      // unpublished geos should still be filtered out
+      const { scopes } = res.locals
+      const { get_published = null, get_unpublished = null } = req.query
+      const params = { get_published, get_unpublished }
+      if (get_published) {
+        params.get_published = get_published === 'true'
+      }
+
+      if (get_unpublished) {
+        params.get_unpublished = get_unpublished === 'true'
+      }
+
+      /* If the user can only read published geos, and all they want is the unpublished geos,
+       * save the trip to the DB and return an array fast.
+       */
+      if (!scopes.includes('geographies:read:unpublished') && params.get_unpublished) {
+        return res.status(200).send([])
+      }
+
+      /* If the user has only the read:published scope, they should not be allowed to see
+       * unpublished geos. If they didn't supply any params, we modify them here so as to
+       * filter only for published geo metadata. We have to monkey with the params here
+       * in a way that we don't for the bulk read of the geographies since we can't filter
+       * the DB results, since metadata has no idea if the geo it's associated with is
+       * published or not.
+       */
+      if (
+        !scopes.includes('geographies:read:unpublished') &&
+        params.get_unpublished === null &&
+        params.get_published === null
+      ) {
+        params.get_published = true
+      }
+      try {
+        const metadata = await db.readBulkGeographyMetadata(params)
+        return res.status(200).send(metadata)
+      } catch (error) {
+        await log.error('failed to read geography metadata', error)
+        /* This error is thrown if both get_published and get_unpublished are set.
+         * To get all geos, neither parameter should be set.
+         */
+        if (error instanceof BadParamsError) {
+          return res.status(400).send({ error })
+        }
+        return res.status(500).send({ error })
+      }
+    }
+  )
+
+  app.get(
+    pathsFor('/geographies/:geography_id'),
+    checkAccess(scopes => {
+      return scopes.includes('geographies:read:published') || scopes.includes('geographies:read:unpublished')
+    }),
+    async (req, res) => {
+      const { geography_id } = req.params
+      try {
+        const geography = await db.readSingleGeography(geography_id)
+        if (!geography.publish_date && !res.locals.scopes.includes('geographies:read:unpublished')) {
+          throw new InsufficientPermissionsError('permission to read unpublished geographies missing')
+        } else {
```

Comment:
Surprised prettier doesn't try to remove the `else` case here, it's unnecessary. You can just have it follow the `if` without the extra syntax.